django_setup_configuration/base.py

```diff
@@ -0,0 +1,168 @@
+
+        field_type = type(field)
+        match field_type:
+            case item if item in basic_field_description.keys():
+                return basic_field_description.get(item)
+            case _:
+                return "No information available"
+
+    def get_concrete_model_fields(self, model) -> Iterator[models.Field]:
+        """
+        Get all concrete fields for a given `model`, skipping over backreferences like
+        `OneToOneRel` and fields that are blacklisted
+        """
+        return (
+            field
+            for field in model._meta.concrete_fields
+            if field.name not in self.excluded_fields
+        )
+
+    def create_config_fields(
+        self,
+        require: tuple[str, ...],
+        exclude: tuple[str, ...],
+        include: tuple[str, ...],
+        model: models.Model,
+        relating_field: models.Field | None = None,
+    ) -> None:
+        """
+        Create a `ConfigField` instance for each field of the given `model` and
+        add it to `self.fields.all` and `self.fields.required`
+
+        Basic fields (`CharField`, `IntegerField` etc) constitute the base case,
+        one-to-one relations (`OneToOneField`) are handled recursively
+
+        `ForeignKey` and `ManyToManyField` are currently not supported (these require
+        special care to avoid recursion errors)
+        """
+
+        model_fields = self.get_concrete_model_fields(model)
+
+        for model_field in model_fields:
+            if isinstance(model_field, OneToOneField):
+                self.create_config_fields(
+                    require=require,
+                    exclude=exclude,
+                    include=include,
+                    model=model_field.related_model,
+                    relating_field=model_field,
+                )
+            else:
+                if model_field.name in self.excluded_fields:
+                    continue
+
+                # model field name could be "api_root",
+                # but we need "xyz_service_api_root" (or similar) for consistency
+                if relating_field:
+                    name = f"{relating_field.name}_{model_field.name}"
+                else:
+                    name = model_field.name
+
+                config_field = ConfigField(
+                    name=name,
+                    verbose_name=model_field.verbose_name,
+                    description=model_field.help_text,
+                    default_value=self.get_default_value(model_field),
+                    field_description=self.get_field_description(model_field),
+                )
+
+                if config_field.name in self.required_fields:
+                    self.config_fields.required.add(config_field)
+
+                # if all_fields is empty, that means we're filtering by blacklist,
+                # hence the config_field is included by default
+                if not self.all_fields or config_field.name in self.all_fields:
```

Comment:
It looks like overkill and can lead to the documentation being incorrect (if a new field is added to the model for example). I think `exclude_fields` is not needed and you need explicitly define all settings, used in configuration. In this case you can loop through settings and not through the model fields and can even have a setting without related model field, which is possible.

Reply:
I removed this part to simplify this, as it clearly is confusing. `excluded_fields` is needed though to exclude things like `id` and other parts of a model that should not show up in the docs.